code/game/objects/items/miscellaneous.dm

```diff
@@ -144,6 +144,109 @@
 	new choice(get_turf(M))
 	to_chat(M, "You hear something crackle from the beacon for a moment before a voice speaks.  \"Please stand by for a message from S.E.L.F. Message as follows: <span class='bold'>Item request received. Your package has been transported, use the autosurgeon supplied to apply the upgrade.</span> Message ends.\"")
 
+/obj/item/choice_beacon/magic
+	name = "beacon of summon magic"
+	desc = "Not actually magical."
+
+/obj/item/choice_beacon/magic/generate_display_names()
+	var/static/list/magic_item_list
+	if(!magic_item_list)
+		magic_item_list = list()
+		var/list/templist = typesof(/obj/item/storage/box/magic) //we have to convert type = name to name = type, how lovely!
+		for(var/V in templist)
+			var/atom/A = V
+			magic_item_list[initial(A.name)] = A
+	return magic_item_list
+
+/obj/item/storage/box/magic
+	name = "Tele-Gloves"
+
+/obj/item/storage/box/magic/PopulateContents()
+	new /obj/item/clothing/gloves/color/white/magic(src)
+
+/obj/item/storage/box/magic/cloak
+	name = "Invisibility Cloak"
+
+/obj/item/storage/box/magic/cloak/PopulateContents()
+	new /obj/item/shadowcloak/magician(src)
+
+/obj/item/storage/box/magic/hat
+	name = "Bottomless Top Hat"
+
+/obj/item/storage/box/magic/hat/PopulateContents()
+	new /obj/item/clothing/head/that/bluespace(src)
+
+/obj/item/clothing/head/that/bluespace //code shamelessly ripped from bluespace body bags, cuz that's basically what this is
+	var/capacity = 2
+	var/maximum_size = 2 //one human, two pets, unlimited tiny mobs, but no big boys like megafauna
+
+/obj/item/clothing/head/that/bluespace/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/M = target
+		M.visible_message("<span class='warning'>[user] starts pulling [src] over [M]'s head!</span>", "<span class='userdanger'>[user] starts pulling [src] over your head!</span>")
+		if(do_after_mob(user, M, (M.health + 100)))//+100 because anyone in crit will be instantly kidnapped otherwise. 
+			if(M == user)
+				M.drop_all_held_items()
```

Comment:
I like how this is before the nodrop check.

Reply:
this proc checks nodrop already, so it won't drop if it has nodrop on, so you'll get sent to nullspace